tests/integration/test_ingress.py

```diff
@@ -32,3 +38,35 @@ async def test_ingress_integration(
         timeout=5,
     )
     assert response.status_code == 200
+
+
+@pytest.mark.abort_on_fail
+async def test_agent_discovery_ingress_integration(
+    model: Model,
+    application: Application,
+    traefik_application: Application,
+    external_hostname: str,
+    jenkins_k8s_agents: Application,
+):
+    """
+    arrange: deploy the Jenkins charm and establish relations via ingress.
+    act: send a request to the ingress in /.
+    assert: the response succeeds.
+    """
+    await application.relate(state.AGENT_RELATION, jenkins_k8s_agents.name)
+    await application.relate(AGENT_DISCOVERY_INGRESS_RELATION_NAME, traefik_application.name)
+
+    await model.wait_for_idle(
+        apps=[application.name, jenkins_k8s_agents.name, traefik_application.name],
+        wait_for_active=True,
+    )
+
+    agent_discovery_url_verified = False
+    for relation in application.relations:
+        if set([jenkins_k8s_agents.name, application.name]) == set(relation.applications):
```

Comment:
```suggestion
        if {jenkins_k8s_agents.name, application.name} == set(relation.applications):
```
Consider this comment as optional